system/src/system_setup.cpp

```diff
@@ -318,11 +318,15 @@ template<typename Config> void SystemSetupConsole<Config>::handle(char c)
     {
         print("Your device MAC address is\r\n");
         IPConfig config = {};
+    #if !HAL_PLATFORM_WIFI    
         auto conf = static_cast<const IPConfig*>(network_config(0, 0, 0));
+    #else
+        auto conf = static_cast<const IPConfig*>(network_config(NETWORK_INTERFACE_WIFI_STA, 0, 0));
+    #endif
         if (conf && conf->size) {
             memcpy(&config, conf, conf->size);
         }
-        const uint8_t* addr = config.nw.uaMacAddr;
+        const uint8_t* addr = conf->nw.uaMacAddr;
```

Comment:
This should be left as `config.`, otherwise if `network_config()` returns `nullptr` you'll have a null pointer dereference here.

Reply:
Is here something wrong?
```
        if (conf && conf->size) {
            memcpy(&config, conf, conf->size);
        }
```
Should be the following, the size of `conf->size` is bigger than `sizeof(config)`?
```
        if (conf && conf->size) {
            memcpy(&config, conf, sizeof(config));
        }
```

Reply:
Or better yet just in case `std::min(sizeof(config), (size_t)conf->size)`

Reply:
Changed as you said.